core/src/main/java/hudson/PluginWrapper.java

```diff
@@ -432,6 +432,20 @@ private String getVersionOf(Manifest manifest) {
     }
 
     /**
+     * Returns the required Java version of this plugin.
+     * @return the required Java version of this plugin.
+     *
```

Comment:
Document what `null` means here? Probably it also makes sense to extrapolate the core dependency to a Java version if undefined (or Just return core Java version once/if #2949 is integrated)

Reply:
I think null is fine here: means the plugin does not specify a version.

Reply:
I think so too. And it's consistent with other similar methods here.